### PR TITLE
TGUI: HonkBot

### PR DESCRIPTION
**Pull request by Oyu07**
## What Does This PR Do
I disgust myself
Honkbot is now TGUI
well, anyways there it is, the glorious honkmother physical manifestation, gift to mankind

## Why It's Good For The Game
MOAR TGUI

## Images of change
image

## Testing
Load master then this PR
Check all settings act the same

## Changelog
:cl:
add: HonkBot TGUI
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/honkbot.dm [hunk 1]
```diff
@@ -69,22 +69,41 @@
...
+/mob/living/simple_animal/bot/honkbot/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "BotHonk", name, 500, 520)
+		ui.open()
+
+/mob/living/simple_animal/bot/honkbot/ui_data(mob/user)
+	var/list/data = ..()
+	return data
+
```
Comment (Gaxeer) [inline code/modules/mob/living/simple_animal/bot/honkbot.dm:84]: redundant proc override. Just remove it

File: code/modules/mob/living/simple_animal/bot/honkbot.dm [hunk 2]
```diff
@@ -69,22 +69,41 @@
...
+/mob/living/simple_animal/bot/honkbot/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "BotHonk", name, 500, 520)
+		ui.open()
+
+/mob/living/simple_animal/bot/honkbot/ui_data(mob/user)
+	var/list/data = ..()
+	return data
```
Comment (Gaxeer) [inline code/modules/mob/living/simple_animal/bot/honkbot.dm:83]: I suppose all of this is just a duplicate of `bot`

File: tgui/packages/tgui/interfaces/BotHonk.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+export const BotHonk = (props, context) => {
+  const { act, data } = useBackend(context);
+  return (
+    <Window resizable>
+      <Window.Content scrollable>
+        <BotStatus />
+      </Window.Content>
+    </Window>
+  );
+};
```
Comment (Gaxeer) [inline tgui/packages/tgui/interfaces/BotHonk.js:15]: redundunt also :D Just use bot ui. That's it

**Review by Gaxeer — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/bot/honkbot.dm (hunk 1 above)
Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/honkbot.dm:84]: it pulls the autopatrol, on off data from the parent, it is still used.

File: code/modules/mob/living/simple_animal/bot/honkbot.dm (hunk 2 above)
Comment (Oyu-De) [reply code/modules/mob/living/simple_animal/bot/honkbot.dm:83]: sorry, wdym by "duplicate of bot"
that entire section is just the tgui backen with data being pulled from bot, as honk bot has no specific settings

File: tgui/packages/tgui/interfaces/BotHonk.js (hunk 3 above)
Comment (Oyu-De) [reply tgui/packages/tgui/interfaces/BotHonk.js:15]: ok I do need help on this, wdym by "bot ui"? what am I missing here?

File: code/modules/mob/living/simple_animal/bot/honkbot.dm (hunk 2 above)
Comment (Gaxeer) [reply code/modules/mob/living/simple_animal/bot/honkbot.dm:83]: I mean, that by calling `..()` you get date from parent and then return it.
But the same behavior will be applied if this override will be absent

File: tgui/packages/tgui/interfaces/BotHonk.js (hunk 3 above)
Comment (Gaxeer) [reply tgui/packages/tgui/interfaces/BotHonk.js:15]: Let me check bot ui fast and I will be back

Comment (Gaxeer) [reply tgui/packages/tgui/interfaces/BotHonk.js:15]: Yep. UI is indeed needed. I am mistaken

Comment (Oyu-De) [reply tgui/packages/tgui/interfaces/BotHonk.js:15]: > Yep. UI is indeed needed. I am mistaken

resolving this one aight?

Comment (Gaxeer) [reply tgui/packages/tgui/interfaces/BotHonk.js:15]: yep

**Review by Gaxeer — APPROVED** [review]

File: code/modules/mob/living/simple_animal/bot/honkbot.dm [hunk 4]
```diff
@@ -69,22 +69,37 @@
...
+		if("power")
+			if(on)
+				turn_off()
+			else
+				turn_on()
+		if("autopatrol")
+			auto_patrol = !auto_patrol
+			bot_reset()
+		if("hack")
+			handle_hacking(usr)
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/bot/honkbot.dm:100]:
```suggestion
			handle_hacking(user)
```

**Review by Sirryan2002 — APPROVED** [review]
just address Henri's suggestion

**Review by Henri215 — APPROVED** [review]
Lewcs good to me now
No need to request a review from Sirryan, he did approve it already.

**Comment by Oyu07** [comment]
> Lewcs good to me now No need to request a review from Sirryan, he did approve it already.

does not look as approved on my end, srry

**Review by S34NW — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**